experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+  _set_even_odd_coefficients!(res)
+  return res
+end
+
+@doc raw"""
+    basis(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the canonical basis of the Clifford algebra $C$.
+"""
+basis(C::CliffordAlgebra) = map(i -> basis(C, i), 1:dim(C))
+
+@doc raw"""
+    gens(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the vector of canonical multiplicative generators of the Clifford algebra $C$.
+"""
+gens(C::CliffordAlgebra) = map(i -> gen(C, i), 1:_dim_qf(C))
+
+@doc raw"""
+    is_commutative(C::CliffordAlgebra) -> Bool
+
+Return `true` if $C$ is commutative and `false` otherwise.
+"""
+is_commutative(C::CliffordAlgebra) = dim(C) == 1 || dim(C) == 2
+
+################################################################################
+#
+#  Element Access
+#
+################################################################################
+@doc raw"""
+    coeff(x::CliffordAlgebraElem, i::Int) -> FieldElem
+
+Return the `i`-th coefficient of the element `x`.
+"""
+coeff(x::CliffordAlgebraElem, i::Int64) = coefficients(x)[i]
+
+getindex(x::CliffordAlgebraElem, i::Int64) = coefficients(x)[i]
+
+################################################################################
+#
+#  Other functionality
+#
+################################################################################
+
+@doc raw"""
+    centroid(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the centroid of $C$. Unless `dim(space(C)) = 0`, it is always two-dimensional,
+so it is returned as a vector containing the basis elements. The first one is the
+multiplicative identity of $C$. The square of the second basis element, if present,
+equals `quadratic_discriminant(C)`.
+"""
+function centroid(C::CliffordAlgebra)
+  if isdefined(C, :centroid)
+    return C.centroid
+  end
+  n = _dim_qf(C)
+  if n == 0
+    C.centroid = [one(C)]
+    return C.centroid
+  end
+  T = orthogonal_basis(space(C))
+  orth_elt = prod(map(i -> sum(map(j -> gen(C, j) * T[i, j], 1:n)), 1:n))
+  orth_elt *= denominator(orth_elt)
+  C.disq = coefficients(orth_elt^2)[1]
+  C.centroid = [one(C), orth_elt]
+end
+
+@doc raw"""
+    quadratic_discriminant(C::CliffordAlgebra) -> FieldElem
+
+Return the quadratic discriminant of $C$ as an element of `base_ring(C)`.
+"""
+function quadratic_discriminant(C::CliffordAlgebra)
+  if isdefined(C, :disq)
+    return C.disq
+  end
+  if dim(space(C)) == 0
+    C.disq = one(base_ring(C))
+  else
+    C.disq = coefficients(centroid(C)[2]^2)[1]
+  end
+end
+
+@doc raw"""
+    disq(C::CliffordAlgebra) -> FieldElem
+
+Alias for `quadratic_discriminant`.
+"""
+disq(C::CliffordAlgebra) = quadratic_discriminant(C)
+
+@doc raw"""
+    center(C::CliffordAlgebra) -> Vector{CliffordAlgebraElem}
+
+Return the center of $C$. It equals `centroid(C)`, if and only if `dim(space(C))`
+is odd. Otherwise it is one-dimensional and returned as a tuple of length one
+containing the multiplicative identity of $C$.
+"""
+function center(C::CliffordAlgebra)
```

Comment:
For algebras this should return a subalgebra together with the embedding. If you want to return a basis, maybe rename it to `center_basis` or `basis_of_center`?

Reply:
Done in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/d4b78c05016fbec7d6ab36f6b18b4b8bec2d084d)